srtcore/common.cpp

```diff
@@ -80,6 +80,55 @@ modified by
 
 #include <srt_compat.h> // SysStrError
 
+
+template<>
+TimeAbs<CK_SYSTEM, TMU_US> TimeAbs<CK_SYSTEM, TMU_US>::now()
+{
+    return TimeAbs<CK_SYSTEM, TMU_US>(CTimer::getTime());
+}
+
+template<>
+void TimeAbs<CK_SYSTEM, TMU_US>::setnow()
+{
+    value = CTimer::getTime();
+}
+
+template<>
+TimeAbs<CK_CPU, TMU_TK> TimeAbs<CK_CPU, TMU_TK>::now()
+{
+    uint64_t value;
+    CTimer::rdtsc(value);
+    return TimeAbs<CK_CPU, TMU_TK>(value);
+}
+
+template<>
+void TimeAbs<CK_CPU, TMU_TK>::setnow()
+{
+    CTimer::rdtsc(value);
+}
+
+
+TimeRel<TMU_TK> TimeConvertTools<TMU_US, TMU_TK>::from(TimeRel<TMU_US> f)
+{
+    return TimeRel<TMU_TK>(f.value * CTimer::getCPUFrequency());
+}
+
+TimeRel<TMU_TK> TimeConvertTools<TMU_US, TMU_TK>::from(TimeRel<TMU_US> f, uint64_t frequency)
+{
+    return TimeRel<TMU_TK>(f.value * frequency);
+}
+
+
+TimeRel<TMU_US> TimeConvertTools<TMU_TK, TMU_US>::from(TimeRel<TMU_TK> f)
+{
+    return TimeRel<TMU_US>(f.value / CTimer::getCPUFrequency());
```

Comment:
As discovered, fix here this type casting to `uin64_t`. And below in
 `TimeConvertTools<TMU_TK, TMU_US>::from(TimeRel<TMU_TK> f, uint64_t frequency)`

Reply:
Wait. Why should I convert this to `uint64_t` while this is `TimeRel`, so the expressed time is relative, that is, it can also be negative?